torch/utils/mkldnn.py

```diff
@@ -178,25 +177,28 @@ def forward(self, x):
         )
 
 
-def to_mkldnn(module):
-    def m_fn(m):
+def to_mkldnn(module, dtype=torch.float):
+    assert dtype in [torch.float, torch.bfloat16], \
+        "MKLDNN only support float or bfloat16 path now"
+
+    def m_fn(m, d):
         if isinstance(m, torch.nn.Linear):
-            return MkldnnLinear(m)
+            return MkldnnLinear(m, d)
         elif isinstance(m, torch.nn.Conv1d):
-            return MkldnnConv1d(m)
+            return MkldnnConv1d(m, d)
         elif isinstance(m, torch.nn.Conv2d):
-            return MkldnnConv2d(m)
+            return MkldnnConv2d(m, d)
         elif isinstance(m, torch.nn.Conv3d):
-            return MkldnnConv3d(m)
+            return MkldnnConv3d(m, d)
         elif isinstance(m, torch.nn.BatchNorm2d) or isinstance(m, torch.nn.BatchNorm3d):
             return MkldnnBatchNorm(m)
```

Comment:
why is dtype ignored for batchnorm? Does it expect its parameters to remain in fp32?

Reply:
Yes, batchnorm's parameters remain in fp32.